### DISCUSSION TITLE
how to use enum as the type for FromQuery binder?

### DISCUSSION THREAD

**Discussion by amirongit** [Neoteroi/BlackSheep#437] [Q&A] [answered]
I get an error like "didn't find a converter, please write one."

is it documented? how can I write a converter? can some one give me a sample?

**Comment by amirongit** [comment]
I tried using Literal, didn't work either; got the same error.
is it possible in V2?

**Reply by sinisaos** [reply, marked as answer]
@amirongit I don't know if you found a way to write a custom converter. I haven't found a way to do that, so I just patch the source code with the `Enum` type in the [bindings converter](https://github.com/Neoteroi/BlackSheep/blob/026897ff2796c2f2074cb368ff9c59a262dc28ad/blacksheep/server/bindings.py#L661) like this
```python
from enum import Enum, EnumType

class SyncBinder(Binder):
    ...
    def _get_default_converter(self, expected_type):
        ...
        if isinstance(expected_type, EnumType):
            return lambda value: value[0] if value else None
```
The usage is like normal FromQuery.
```python
import typing as t
from enum import Enum, EnumType

from blacksheep import Application, FromQuery, get
from blacksheep.server.openapi.v3 import OpenAPIHandler
from openapidocs.v3 import Info


class OrderEnum(str, Enum):
    ASC = "asc"
    DESC = "desc"


app = Application()


docs = OpenAPIHandler(info=Info(title="Example API", version="0.0.1"))
docs.bind_app(app)


@get("/")
def home(order_by: t.Optional[FromQuery[OrderEnum]] = None):
    try:
        return order_by.value
    except AttributeError:
        pass
```
I hope that is useful.
